File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/UsedeskFile.java
package ru.usedesk.chat_sdk.external.entity;

import android.support.annotation.NonNull;

public class UsedeskFile {

    private static final String IMAGE_TYPE = "image/";

    private final String content;
    private final String type;
    private final String size;
    private final String name;

    public UsedeskFile(@NonNull String content, @NonNull String type, String size, @NonNull String name) {
        this.content = content;
        this.type = type;
        this.size = size;
        this.name = name;
    }

    public String getContent() {
        return content;
    }

    public String getType() {
        return type;
    }

    public String getSize() {
        return size;
    }

    public String getName() {
        return name;
    }

    public boolean isImage() {
        return type.startsWith(IMAGE_TYPE) || endsLikeImage();
    }

    private boolean endsLikeImage() {
        return name.endsWith(".png") || name.endsWith(".jpg") ||
                name.endsWith(".bmp") || name.endsWith(".jpeg");
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/_entity/UsedeskFile.kt
package ru.usedesk.chat_sdk.data._entity

import android.webkit.MimeTypeMap
import com.google.gson.Gson
import ru.usedesk.chat_sdk.entity.UsedeskFileInfo

data class UsedeskFile(
        val content: String,
        val type: String,
        val size: String,
        val name: String
) {
    fun isImage(): Boolean {
        return type.startsWith(UsedeskFileInfo.IMAGE_TYPE)
    }

    fun serialize(): String = Gson().toJson(this)

    companion object {
        fun deserialize(json: String): UsedeskFile = Gson().fromJson(json, UsedeskFile::class.java)

        fun create(content: String,
                   type: String?,
                   size: String,
                   name: String): UsedeskFile {
            val mimeType = if (type?.contains('/') == true) {
                type
            } else {
                MimeTypeMap.getSingleton().getMimeTypeFromExtension(type ?: "")
            } ?: ""
            return UsedeskFile(content, mimeType, size, name)
        }
    }
}